**Исключения**  
1. Наследуются от класса Throwable
2. Виды
- Проверяемые
- Не проверяемые

**Не проверяемые**  
Иерархия: **Throwable** -> **Error**
Некоторые классы наследуемые от **Error**
- ThreadDeath
- AssertionError
- VMError
- Internal Error
- Unknown Error
- OutOfMemoryError
- StackOverflowError  

Исключительные ситуации Error возникают только во время выполнения программы.
Такие Исключения связаны с серьезными ошибками такими как, переполнение стека.
**Не не подлежат исправлению и не м.б обработаны приложением**
  
**Проверяемые исключения**  
Проверяемые исключения должны быть обработаны в методе, который может их генерировать,
или должны быть включены в **throws-** список метода для дальнейшей обработки в 
вызывающих методах.

Таблица классов непроверяемых исключений наследуемых от класса RuntimeException  

| Исключение                      | Значение                                                     |
|---------------------------------|--------------------------------------------------------------|
| ArithmeticException             | Арифметическая ошибка деление на ноль и др                   |
| ArrayIndexOutOfBoundsException  | Индекс массива находится вне его границ                      |
| ArrayStoreException             | Назначение элементу массива несовместимого типа              |
| ClassCastException Недопустимое | приведение типов                                             |
| ConcurrentModificationException | Некорректный способ модификации коллекции                    |
| IllegalArgumentException        | При вызове метода использован некорректный аргумент          |
| IllegalMonitorStateException    | Незаконная операция монитора на разблокированном объекте     |
| IllegalStateException           | Среда или приложение находятся в некорректном состоянии      |
| IllegalThreadStateException     | Требуемая операция не совместима с текущим состоянием потока |
| IndexOutOfBoundsException       | Некоторый тип индекса находится вне границ коллекции         |
| NegativeArraySizeException      | Попытка создания массива с отрицательным размером            |
| NullPointerException            | Недопустимое использование ссылки на null                    |
| NumberFormatException           | Невозможное преобразование строки в числовой формат          |
| StringIndexOutOfBoundsException | Попытка индексации вне границ строки                         |
| MissingResourceException        | Отсутствие файла ресурсов properties или имени ресурса в нем |
| EnumConstantNotPresentException | Несуществующий элемент перечисления                          |
| UnsupportedOperationException   | Встретилась неподдерживаемая операция                        |

**Блок finally**  
Часто используется для закрытия файлов и освобождения других ресурсво, захваченных для временного использования в начале
выполнения метода. **Код блока выполняется даже если перед ним были выполнены такие инструкции, как throws, return, break,
continue**
